src/include/duckdb/function/function.hpp

```diff
@@ -168,6 +173,8 @@ class BaseScalarFunction : public SimpleFunction {
 	FunctionStability stability;
 	//! How this function handles NULL values
 	FunctionNullHandling null_handling;
+	//! Whether or not this function can throw an error
+	FunctionErrors errors;
```

Comment:
Can we use this in `BoundFunctionExpression::CanThrow()`?

Reply:
And perhaps add a test with some functions that throw to ensure they are not re-ordered by the optimizer, similar to the other issues we encountered?